### PR TITLE
bpo-38753: AsyncMock added in version 3.8

### PR DESCRIPTION
**Pull request by belm0**
https://bugs.python.org/issue38753

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@belm0

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by belm0** [comment]
CLA submitted
needs backport to 3.8 🙏

File: Doc/library/unittest.mock.rst
Comment (tirkarthi) [inline Doc/library/unittest.mock.rst]: I would suggest moving this to somewhere near the class definition signature at https://github.com/python/cpython/pull/17102/files#diff-4d545f72ef0b42afd6a20165e4374004L855 before the first paragraph of doc so that users can quickly know without scrolling down.

NEWS entry is not required for this since it's a documentation change. We can add "skip-news" label to keep CI happy.

Comment (belm0) [reply Doc/library/unittest.mock.rst]: I've moved it to just ahead of the method definitions, following e.g. https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack

**Comment by belm0** [comment]
I think bpo associated this PR with my older bpo account, which doesn't have a CLA.  I'll submit a CLA for that account as well...

https://bugs.python.org/pull_request16608

**Comment by belm0** [comment]
please add 3.8 backport tag

**Review by tirkarthi — APPROVED** [review]
I think adding it before "An asynchronous version of :class:`Mock`. The :class:`AsyncMock`" paragraph would be ideal. I will leave it to the core dev merging this. It will be backported to 3.8 once merged. Thanks :)

**Review by aeros — APPROVED** [review]
Thanks for the PR @belm0.

@tirkarthi 
> I think adding it before "An asynchronous version of :class:Mock. The :class:AsyncMock" paragraph would be ideal.

The most common formats I've seen across the docs are:

1)
```
..type:: callable(*args, **kwargs)

Summary

Example(s)

Version added
```

2)
```
..type:: callable(*args, **kwargs)

Summary

Version added

Example(s)
```

The most common one seems to be (1), which is essentially the current placement of the `versionadded` added notice in the PR, so I think the current location is fine. For an example of (1), see https://docs.python.org/3/library/asyncio-task.html#asyncio.run.

**Comment by belm0** [comment]
is CLA process broken?  I submitted doc about 4 days ago.

**Comment by aeros** [comment]
@belm0 
> is CLA process broken? I submitted doc about 4 days ago.

Try manually updating the GitHub CLA status via https://check-python-cla.herokuapp.com/, I recall having to do that for my first PR.

**Comment by lisroach** [comment]
LGTM, thanks for adding this :)

**Review by lisroach — APPROVED** [review]

**Pull request merged**
